MStokluska/integreatly-operator
# MGDAPI-4744 - feat: native 3scale sts support

# Issue link

https://issues.redhat.com/browse/MGDAPI-4744

# What

Create s3 in the sts secret format expected by 3scale operator for STS clusters

# Verification steps



* Verify unit tests passes
```
make test/unit
```
* Provision sts cluster
* Create s3 role for 3scale 
```
CLUSTER_NAME=<YOUR_CLUSTER_NAME>

get_cluster_id() {
    local CLUSTER_ID=$(ocm get clusters --parameter search="name like '%$CLUSTER_NAME%'" | jq '.items[].id' -r)
    echo "$CLUSTER_ID"
}

get_oidc_provider() {
    local OIDC_PROVIDER=$(aws iam list-open-id-connect-providers | jq -r --arg CLUSTER_ID "$(get_cluster_id)" '.OpenIDConnectProviderList[] | select(.Arn | endswith($CLUSTER_ID)).Arn')
    echo "$OIDC_PROVIDER"
}

    cat <<EOM >"chfan-s3-amp.json"
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Principal": {
          "Federated": [
              "$(get_oidc_provider)"
          ]
      },
      "Action": ["sts:AssumeRoleWithWebIdentity"]
    }
  ]
}
EOM

    aws iam delete-role-policy --role-name chfan-s3-amp --policy-name chfan-s3-amp-policy || true
    aws iam delete-role --role-name chfan-s3-amp || true

    aws iam create-role --role-name chfan-s3-amp --assume-role-policy-document "file://chfan-s3-amp.json" || true
    cat <<EOM >"chfan-s3-amp-policy.json"
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "s3:*"
            ],
            "Resource": "*"
        }
    ]
}
EOM
    aws iam put-role-policy --role-name chfan-s3-amp --policy-name chfan-s3-amp-policy --policy-document "file://chfan-s3-amp-policy.json" || true
```
* Prepare cluster 
```
3SCALE_ROLE_ARN=<CREATED_ARN> make cluster/prepare/local
```
* Install RHOAM until the 3scale blobstorage is provisioned by CRO
```
make code/run
```
* Stop the operator
* Update the `threescale-blobstorage-rhoam` secret is the rhoam operator namespace to the following values:
  * bucketName - chfan-test
  * bucketRegion - eu-west-2
* Run the operator
* Verify RHOAM installs successfully
* Verify the developer developer portal is up and accessible
* Uninstall RHOAM
